src/Microsoft.Health.Dicom.Core/Features/Query/QueryService.cs

```diff
@@ -151,11 +151,12 @@ private static bool ReturnNewTagDefaults(int? version)
             else
             {
                 Dictionary<DicomIdentifier, SeriesResult> map = seriesComputedResults.ToDictionary<SeriesResult, DicomIdentifier>(a => new DicomIdentifier(a.StudyInstanceUid, a.SeriesInstanceUid, default));
-                instanceMetadata = instanceMetadata.Select(x =>
-                {
-                    var ds = new DicomDataset(x);
-                    return ds.AddOrUpdate(map[new DicomIdentifier(x.GetSingleValue<string>(DicomTag.StudyInstanceUID), x.GetSingleValue<string>(DicomTag.SeriesInstanceUID), default)].DicomDataset);
-                });
+                instanceMetadata = instanceMetadata.Where(x =>
```

Comment:
Not your change, can you materialize all the selects in line 141, 149 and 167

Reply:
done.